Update release doc (#651)
CesiumGS/cesium-unity: Documentation~/release-guide.md

<!--! [TOC] -->

## Prepare Cesium Native for Release

[cesium-native](https://github.com/CesiumGS/cesium-native) is used as a git submodule of cesium-unity. So the first step in releasing Cesium for Unity is to make sure that Cesium Native is ready for release.

In the cesium-native repo:

1. Verify that CI is passing on all platforms. Fix it if not.
2. Verify that `CHANGES.md` is complete and accurate.
   - Give the header of the section containing the latest changes an appropriate version number and date.
   - Diff main against the previous released version. This helps catch changes that are missing from the changelog, as well as changelog entries that were accidentally added to the wrong section.
3. Set the `version` property in `package.json`.
4. Set the `VERSION` property passed to the `project()` function in `CMakeLists.txt`.
5. Commit these changes. You can push them directly to `main`.

## Prepare Cesium for Unity for Release

1. Update the cesium-native submodule reference to point to the latest commit that you pushed above.
   - Enter the `native~/extern/cesium-native` directory and pull the latest changes and checkout the main branch as normal.

>    - To update the submodule reference, from the cesium-unreal root, run: `git add native~/extern/cesium-native`

Reviewer: ```suggestion
   - To update the submodule reference, from the cesium-unity root, run: `git add native~/extern/cesium-native`
```

4. Commit these changes. You can push them directly to `main`.
5. Before continuing, verify that CI passes for all platforms and Unreal Engine versions.

## Optimistically upload the release packages to the GitHub releases page

Cesium for Unity release package is fairly large, so it can take awhile to upload it to the GitHub Releases page. For that reason, it is helpful to start the process now, even though we haven't tested the release yet.

1. Go to https://github.com/CesiumGS/cesium-unity/actions.
2. Click the most recent build of the `main` branch (it should be near the top). This should be the build for the commit you pushed in the previous section. If it's still running, wait. If it failed, you'll need to fix it.
3. Scroll down to **Artifacts**.
4. Download the artifact named "Combined Package".
5. Extract the `com.cesium.unity-1.21.0.tgz` (or similar) from the `Combined Package.zip` file. On macOS this is a pain to do, because Finder's unzip insists on also extracting the contents of the .tgz file. The best approach is probably to use `unzip` from the command-line.
6. Create a new release by visiting https://github.com/CesiumGS/cesium-unity/releases/new.

> 7. We haven't created a tag for this release yet, so leave that blank for now.

Reviewer: For verb-first symmetry with the rest of the list:

```suggestion
7. Leave the release tag blank for now, as we haven't created it yet.
```

8. Set `Release Title` to "Cesium for Unity v1.21.0", updating the version number as appropriate.
9. Copy the "release notes" section from a previous release, which you can find by visiting https://github.com/CesiumGS/cesium-unity/releases/latest and clicking the Edit button. But be careful not to save!
10. Update the version numbers as appropriate in the top section. Replace the changelog section with the actual changelog entries from this release. Copy it from `CHANGES.md`.
11. Upload the .tgz file by dragging it into the "Attach binaries" box.
12. Click Save Draft. Be careful not to publish it yet.

> 

Reviewer: Is it worth also adding Steps 13 + 14 from the Cesium for Unreal guide here, too?

Author: Yeah I think the best we can do here is make build process output the sha256 of the tgz file before it includes it in the artifact. Then we could compare the value in the log to the sha256 value GitHub shows for the asset to make sure nothing sus has happened.

> If the uploaded packages are later found to have problems during testing, this step will need to be repeated.

Reviewer: ```suggestion
If the uploaded package is later found to have problems during testing, this step will need to be repeated.
```

## Test the release candidate

1. Launch the project in the Unity Editor. When prompted, click `Enter Safe Mode`, since all of the Cesium objects and scripts will be missing because the specified package version doesn't exist yet.
2. Go to Window -> Package Manager. Click on the add icon in the top left corner and select "Add package from tarball". Navigate to where you extracted the tarball, then select it and click "Open".
3. Assuming there are no more compilation errors, Unity should automatically exit Safe Mode once it processes the package. You should see Cesium for Unity installed in the Package Manager window.
4. Open each scene in Assets -> CesiumForUnitySamples -> Scenes and verify it works correctly:
   * Does it open without crashing?
   * Does it look correct?
   * Press Play. Does it work as expected? The billboard in each level should give you a good idea of what to expect.
5. Test on other platforms and other versions of Unity if you can. If you can't (e.g., you don't have a Mac), post a message on Slack asking others to give it at least a quick smoke test.

If all of the above goes well, you're ready to release Cesium for Unity.

## Publish the release on GitHub

> 1. Tag the cesium-native release, and push the tag. Be sure you're tagging the exact commit that cesium-unity is using.

Reviewer: ```suggestion
1. Tag the cesium-native release if you haven't already, and push the tag. Be sure you're tagging the exact commit that cesium-unity is using.
```
